### ISSUE TITLE
HiDPI display support

### ISSUE THREAD

**Issue by umpirsky** [clementine-player/Clementine#5338]
On HiDPI displays artist tags and left sidebar are not displayed correctly, see screenshot below:

![clementine](https://cloud.githubusercontent.com/assets/208957/14597611/217814c6-054e-11e6-86e6-a1157541ccba.png)

### System information

- Operating System: Ubuntu 15.10
- Clementine version: 1.3
- Hardware: Asus Zenbook UX301L
- Display resolution: 2560 x 1440

**Comment by Chocobozzz** [comment]
Can you test the Qt5 branch with Qt 5.6 and make another screenshot?

**Comment by narunlifescience** [comment]
I think this happens because of hard coded font size values.. QT5 may not fix it i guess..

**Comment by umpirsky** [comment]
@Chocobozzz Following [compiling from source](https://github.com/clementine-player/Clementine#compiling-from-source) from readme.

Failed to cmake:

```
$ cmake ..
-- The C compiler identification is GNU 5.2.1
-- The CXX compiler identification is GNU 5.2.1
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.28") 
CMake Error at CMakeLists.txt:37 (find_package):
  By not providing "FindQt5.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "Qt5", but
  CMake did not find one.

  Could not find a package configuration file provided by "Qt5" (requested
  version 5.6.0) with any of the following names:

    Qt5Config.cmake
    qt5-config.cmake

  Add the installation prefix of "Qt5" to CMAKE_PREFIX_PATH or set "Qt5_DIR"
  to a directory containing one of the above files.  If "Qt5" provides a
  separate development package or SDK, be sure it has been installed.


-- Configuring incomplete, errors occurred!
See also "/home/umpirsky/Projects/Clementine/bin/CMakeFiles/CMakeOutput.log".
```

[CMakeOutput.log](https://gist.github.com/umpirsky/cf5bdcffc633cf59e66475c9d916a919)

**Comment by Brli** [comment]
![screenshot_20160420_080308](https://cloud.githubusercontent.com/assets/4042566/14659404/62d646f0-06ce-11e6-8fdd-20dac9158d62.png)
The Qt5  f182562 version still the same

**Comment by ermolaev** [comment]
info about hidpi in qt http://blog.davidedmundson.co.uk/blog/kde_apps_high_dpi

I use qBittorrent and if run as `QT_DEVICE_PIXEL_RATIO=2 qbittorrent` it work fine on 4k display

**Comment by umpirsky** [comment]
@ermolaev I tried `QT_DEVICE_PIXEL_RATIO=2 clementine`, same result.

**Comment by ermolaev** [comment]
@umpirsky it will be work after porting clementine to qt 5.6
waiting... or download qt 5.6 and build from source

**Comment by umpirsky** [comment]
@ermolaev Oh, ok, thanks for the info.

**Comment by sbrl** [comment]
I'm experiencing interface scaling issues on my Dell Inspiron 5567. My screen is 1920 by 1080 pixels, with a physical size of 508mm by 285mm, according to `xdpyinfo  | grep 'dimensions:'`.

The text is really small, and the all the interface icons are equally so.

**Comment by siddharthist** [comment]
@sbrl The quickest way to get better HiDPI support would be to pitch in on issue https://github.com/clementine-player/Clementine/issues/5334 (if you've got a Windows or OSX computer).

**Comment by sbrl** [comment]
@siddharthist No, sorry! I don't have a Windows or a Mac machine. My laptop is currently running Linux - more specifically Ubuntu 16.04.1.

**Comment by UnicodeAlt255** [comment]
For me, I haven't had hidpi issues except for the fact that the control buttons icons (e.g. repeat, shuffle, play, skip) are really tiny, even smaller than a single digit of the track length number.

**Comment by opennomad** [comment]
It's not a fix, but seems to work as workaround for me on Ubuntu: https://superuser.com/questions/950794/per-application-window-scaling-in-xorg-for-high-dpi-display/1189909#1189909

**Comment by CottonEaster** [comment]
A workaround is not ideal, especially in this case where, at least on my system, _some_ parts of Clementine are scaled correctly. Still, if the newer version of QT fixes the problem, then I suppose we can wait for that version to reach us (or compile it from source).

**Comment by sbrl** [comment]
`run_scaled` has some stability issues, unfortunately. What's the status of this?

**Comment by axelsimon** [comment]
As this is the issue with most work on it, i've marked the other related issues as duplicates of this one (even if they all predate this one).

**Comment by PowerSnail** [comment]
I compiled Clementine with Qt 5.14, while adding the following lines:

```cpp
  QCoreApplication::setAttribute(Qt::AA_UseHighDpiPixmaps);
  QCoreApplication::setAttribute(Qt::AA_EnableHighDpiScaling);
```

The result seems pretty promising:

![Screenshot from 2020-02-20 22-15-33@2x](https://user-images.githubusercontent.com/7428673/75009169-e51ad500-5471-11ea-851c-0c9441270165.png)

I haven't found any placement problem or scaling problem yet.

- System: Elementary OS 5.1
- Qt 5.14
- Clementine: git master
- Monitor: 4k resolution, 2x scaling

**Comment by sbrl** [comment]
@PowerSnail I compiled Clementine with Qt 5.14, while adding the following lines:

> QCoreApplication::setAttribute(Qt::AA_UseHighDpiPixmaps);
> QCoreApplication::setAttribute(Qt::AA_EnableHighDpiScaling);

PR? :P

**Comment by PowerSnail** [comment]
> PR? 

Just made one. #6592

**Comment by sbrl** [comment]
Awesome <3

Can't wait for the next release!

**Comment by northnose** [comment]
I see that these lines are merged, but when I build from source (2020-03-24 a22af31e6eaa5195f63717163c09e25d765e1fb5), I don't see any difference. This is on Fedora 31, KDE 5.67, Qt 5.13.2, with "Force font DPI" set to 192.

![Screenshot_20200327_215536](https://user-images.githubusercontent.com/4389659/77815209-fc9f3c00-7075-11ea-9f5b-a35f78bae2fd.png)

**Comment by PowerSnail** [comment]
I was using scale = 2 for monitor setting.

My guess is that under the hood, Qt auto-scale widgets/layouts/pixmaps based on device pixel ratio, whose calculation isn't affected by font DPI. As a result, when you set "Force font DPI", some widgets are resized because their sizes are calculated with regard to text. Others are not resized, because they are not aware of the scaling. 

But you can try setting the scale for monitor and not forcing font DPI; if that works, then my guess would be confirmed.

**Comment by northnose** [comment]
Ah, my bad. I didn't realize that the font DPI and the display scaling are different things altogether. Thanks for the pointer. Would it make sense to file a separate issue about font DPI?

In the meantime, if anyone else has this issue, I can work around it (albeit with small text in Clementine) with a simple patch to clementine.desktop:

```
-Exec=clementine %U
+Exec=env QT_FONT_DPI=96 clementine %U
```

**Comment by duncanawoods** [comment]
@northnose 

Does this work with a snap install? I can't see a clementine.desktop in `/home/xxx/.local/share` and if I create one I don't see any change.

**Comment by endlisnis** [comment]
I still see this issue running v1.4 rc1 on Linux Mint 20.1.
